drivers/include/drivers/UnbufferedSerial.h

```diff
@@ -143,6 +143,33 @@ class UnbufferedSerial:
         return 0;
     }
 
+    /** Enable or disable input
+     *
+     * Control enabling of device for input. This is primarily intended
+     * for temporary power-saving; the overall ability of the device to operate
+     * for input and/or output may be fixed at creation time, but this call can
+     * allow input to be temporarily disabled to permit power saving without
+     * losing device state.
+     *
+     *  @param enabled      true to enable input, false to disable.
+     *
+     *  @return             0 on success
```

Comment:
What would it return on failure?

Reply:
@kjbracey-arm there is nothing in place in the override function for returning an error code on failure

Reply:
The base class doesn't return anything either, I'm not sure how you can be sure the operation has succeeded.

Reply:
No, but the current implementation does not have to define the API.

I don't recall whether it's inherently impossible for `SerialBase` to fail or if it's just failing to report failures.

Maybe you might want to make it public API that this call can't fail in `UnbufferedSerial`, unlike in `FileHandle` more generally. Calls that can't fail do make your life easier. But it needs a bit more thought before locking down API than "it currently doesn't". And you'd need to say that explicitly, not just "0 on success" leaving the "what about failure" question open.

Reply:
Unfortunately, I don't have enough familiarity with the API to make a call here. @hugueskamba what's your view on this?

Reply:
Though the method overrides the `FileHandle` implementation, the fact it was implemented using the implementation of the `SerialBase` class (which in its current state does not fail) suggested that the documentation was incorrect.

However, since the `SerialBase` implementation's fate is still up in the air as indicated below, it makes sense to not lock the API.